### ISSUE TITLE
Is there any update of segmental-rnn example?

### ISSUE THREAD

**Issue by Lavender0225** [clab/dynet#1113]
Hi,
I notice that segmental-rnn still uses boost... But now dynet has no boost part. So is there any update of this example?

By the way...
I cannot compile example XOR (train_xor.cc) for test. I'm using osx 10.13.1. 
I have no idea what should I do....

The errors:

`Undefined symbols for architecture x86_64:
  "dynet::initialize(int&, char**&, bool)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::TextFileSaver::save(dynet::ParameterCollection const&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::TextFileSaver::TextFileSaver(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, bool)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::TextFileSaver::~TextFileSaver()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::TextFileLoader::populate(dynet::ParameterCollection&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::TextFileLoader::TextFileLoader(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::TextFileLoader::~TextFileLoader()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::default_device", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ComputationGraph::forward(dynet::Expression const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ComputationGraph::backward(dynet::Expression const&, bool)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ComputationGraph::ComputationGraph()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ComputationGraph::~ComputationGraph()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::squared_distance(dynet::Expression const&, dynet::Expression const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ParameterCollection::add_parameters(dynet::Dim const&, float, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, dynet::Device*)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ParameterCollection::ParameterCollection()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ParameterCollection::~ParameterCollection()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::tanh(dynet::Expression const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::input(dynet::ComputationGraph&, float const*, dynet::Device*)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::input(dynet::ComputationGraph&, dynet::Dim const&, std::__1::vector<float, std::__1::allocator<float> > const*, dynet::Device*)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::Trainer::update()", referenced from:
      _main in train_xor-41f71c.o
  "dynet::Trainer::~Trainer()", referenced from:
      dynet::SimpleSGDTrainer::~SimpleSGDTrainer() in train_xor-41f71c.o
  "dynet::as_scalar(dynet::Tensor const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::parameter(dynet::ComputationGraph&, dynet::Parameter)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::operator*(dynet::Expression const&, dynet::Expression const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::operator+(dynet::Expression const&, dynet::Expression const&)", referenced from:
      _main in train_xor-41f71c.o
  "dynet::ComputationGraph::print_graphviz() const", referenced from:
      _main in train_xor-41f71c.o
  "vtable for dynet::SimpleSGDTrainer", referenced from:
      dynet::SimpleSGDTrainer::SimpleSGDTrainer(dynet::ParameterCollection&, float) in train_xor-41f71c.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for dynet::Trainer", referenced from:
      dynet::Trainer::Trainer(dynet::ParameterCollection&, float) in train_xor-41f71c.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)`

And my sublime build configure is
`
    "cmd": ["clang++", "${file}", "-std=c++11", "-stdlib=libc++", "-o", "${file_path}/${file_base_name}", 
    	"-I/Users/Jingzhang/study/deep_learning/dynet/dynet",
    	"-I/usr/local/include/eigen3",
        "-I/usr/local/Cellar/boost/1.65.1",
        "-L//Users/Jingzhang/study/deep_learning/dynet/dynet/build/dynet"],
    "file_regex": "^(..[^:]*):([0-9]+):?([0-9]+)?:? (.*)$",
    "working_dir": "${file_path}",
    "selector": "source.c, source.c++, source.cxx, source.cc",
`

**Comment by neubig** [comment]
No, at the moment we're not planning on maintaining this extensively. We'd welcome a PR fixing any problems though.
